src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/DesginTimeODP.java

```diff
@@ -0,0 +1,121 @@
+import java.io.IOException;
+
+/**
+ * DesginTimeODP.
+ */
+public class DesginTimeODP {
+
+  @Then("Connection is established")
+  public void connectionIsEstablished() throws InterruptedException {
+    CdfGcsActions.getSchema();
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 20000);
+    wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(By.xpath("//*[@placeholder=\"Field name\"]"), 2));
+    Assert.assertEquals(true, CDAPUtils.schemaValidation());
+  }
+
+  @When("LoadProp {string} {string} {string} {string} {string} {string} {string} {string} {string}")
+  public void loadConnection(String client, String asHost, String msServ, String systemID, String dsName,
+                             String gcsPath, String splitRow, String pkgSize, String lgrp)
+    throws IOException, InterruptedException {
+    ODPActions.getODPProperties();
+    ODPActions.enterLoadConnectionProperties(
+      client, asHost, msServ, systemID, lgrp, dsName, gcsPath, splitRow, pkgSize);
+  }
+
+  @When("User has selected Sap client macro to configure")
+  public void userHasSelectedSapClientMacroToConfigure() {
+    ODPLocators.macros.get(0).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.sapClient, "${client}");
+  }
+
+  @Then("User is validate without any error")
+  public void userIsValidateWithoutAnyError() throws InterruptedException {
+    ODPLocators.validateButton.click();
+    Assert.assertTrue(ODPLocators.pluginValidationSuccessMessage.isDisplayed());
+  }
+
+  @When("User has selected Sap language macro to configure")
+  public void userHasSelectedSapLagMacroToConfigure() {
+    ODPLocators.macros.get(1).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.language, "${lang}");
+  }
+
+  @When("User has selected Sap server as host macro to configure")
+  public void userHasSelectedSapServerAsHostMacroToConfigure() {
+    ODPLocators.macros.get(2).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.sapApplicationServerHost, "${host}");
+  }
+
+  @When("User has selected System Number macro to configure")
+  public void userHasSelectedSystemNumberMacroToConfigure() {
+    ODPLocators.macros.get(3).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.systemNumber, "${sysnr}");
+  }
+
+  @When("User has selected datasource macro to configure")
+  public void userHasSelectedDatasourceMacroToConfigure() {
+    ODPLocators.macros.get(5).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.dataSourceName, "${dsname}");
+  }
+
+  @When("User has selected gcsPath macro to configure")
+  public void userHasSelectedGcsPathMacroToConfigure() {
+    ODPLocators.macros.get(10).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.gcsPath, "${gcs}");
+  }
+
+  @When("User has selected Sap msHost macro to configure")
+  public void userHasSelectedSapMsHostMacroToConfigure() {
+    ODPLocators.macros.get(2).click();
+    SeleniumHelper.replaceElementValue(ODPLocators.msHost, "${msHOst}");
```

Comment:
${msHost} ?

Reply:
Thats a macro variable which we have taken.

Reply:
done